python/ops.py

```diff
@@ -13,12 +13,37 @@
 # limitations under the License.
 """FFCC TensorFlow ops."""
 import math
-import sys
-
+import itertools
 import numpy as np
 import tensorflow.compat.v1 as tf
 
 
+
+EPS = tf.constant(1e-9, dtype=tf.float32)
+
+
+def edge_kernel():
+  """Construct a set of 3x3 kernels for edge computation."""
+  filters = np.zeros((8, 3, 3, 1))
+  filters[:, 1, 1, :] = 1
+  offsets = [-1, 0, 1]
+  j = 0
+  for i, (dx, dy) in enumerate(itertools.product(offsets, repeat=2)):
+    if dx == 0 and dy == 0:
+      j += 1
+      continue
+    filters[i - j, 1 + dx, 1 + dy, 0] = -1
+  return tf.constant(filters, dtype=tf.float32)
+
+
+edge_filters = edge_kernel()
```

Comment:
Avoid module level variables, especially as these are tf v1 tensors that need to be created during a Session. You might get errors later during training. Just define them where they will be used.

Reply:
Fixed -- 'edge_filters' are created before using them in 'local_absolute_deviation'.